api/base/serializers.py

```diff
@@ -99,11 +99,12 @@ def __init__(self, **kwargs):
 
     # Overrides CharField
     def to_internal_value(self, data):
-        request = self.context['request']
-        if request.method in utils.UPDATE_METHODS and not utils.is_bulk_request(request):
-            id_field = getattr(self.root.instance, self.source, '_id')
-            if id_field != data:
-                raise Conflict()
+        request = self.context.get('request')
+        if request:
+            if request.method in utils.UPDATE_METHODS and not utils.is_bulk_request(request):
+                id_field = getattr(self.root.instance, self.source, '_id')
```

Comment:
I don't understand why the default is `_id` here. Isn't `data` going to be an ID, not a field name? If that's the case, we're comparing a GUID to `_id`, which doesn't seem right.

That said, I see that this isn't new code, so we can leave it alone. I would still like to understand this logic, though.